pkg/image/fetcher.go

```diff
@@ -136,8 +139,21 @@ func (f *Fetcher) fetchDaemonImage(name string) (imgutil.Image, error) {
 	return image, nil
 }
 
-func (f *Fetcher) fetchRemoteImage(name string) (imgutil.Image, error) {
-	image, err := remote.NewImage(name, f.keychain, remote.FromBaseImage(name))
+func (f *Fetcher) fetchRemoteImage(name string, target *dist.Target, multiarch bool) (imgutil.Image, error) {
+	var (
+		image imgutil.Image
+		err   error
+	)
+
+	if target == nil {
+		image, err = remote.NewImage(name, f.keychain, remote.FromBaseImage(name))
+	} else {
+		if multiarch {
+			image, err = remote.NewImage(name, f.keychain, remote.FromBaseImage(name), remote.WithDefaultPlatform(imgutil.Platform{OS: target.OS, Architecture: target.Arch}), remote.SaveWithDigest())
```

Comment:
Could we add a comment here? Why do we need `remote.SaveWithDigest()` when multiarch is true?

Reply:
I need to double check this logic. 

When I did the PoC, and suppose you want to publish `linux/amd64` and `linux/arm64` buildpack packages, then you provided:

`pack buildpack package some-buildpack-index` 

I need to use some intermediate name to refer to each manifest when adding them to the Image Index.  See the `Create Manifest` command. 

At that moment, I used the digest to save the image like: `some-buildpack-index:digest` I think and then I used that to add it into the index.

Reply:
This is the [PR](https://github.com/buildpacks/imgutil/pull/251) created on imgutil. Again, this was just a hack for the RFC

Reply:
I do not need it, I just removed this logic in this [commit](https://github.com/buildpacks/pack/pull/2086/commits/9c8a2eef283853abf9ffa4494b6077c05a8d57c5)